### ISSUE TITLE
UPDATE TAGS

### ISSUE THREAD

**Issue by ericbrunner** [adrianhall/develop-mobile-apps-with-csharp-and-azure#27]
Does that code work (HttpMethod.Patch) to add custom Tags from my Mobile App Backend?

```
   // Update Tags with NH
   var response = await client.InvokeApiAsync<DeviceInstallation, DeviceInstallation>(
   $"/push/installations/{client.InstallationId}",
   installation,
   HttpMethod.Patch,
   new Dictionary<string, string>());
```

OR 

Does that [UpdateTagsController ](https://github.com/Azure-Samples/app-service-mobile-dotnet-backend-quickstart/blob/master/todolist_completeService/Controllers/UpdateTagsController.cs)still work with App Service PUSH or is a call to 

```
   // Add the requested tag to the installation.
   await hubClient.PatchInstallationAsync(Id, updates); 
```

**deprecated** !?

**Comment by adrianhall** [comment]
The /push/installations method is a createOrUpdateInstallation call on the backend and it's a POST (not a PATCH).

**Comment by ericbrunner** [comment]
Its a **PUT** isn't it like in your ZUMO BOOK code:

```
  // Register with NH
  var response = await client.InvokeApiAsync<DeviceInstallation, DeviceInstallation>(
                    $"/push/installations/{client.InstallationId}",
                    installation,
                    HttpMethod.Put,
                    new Dictionary<string, string>());
```

**Comment by adrianhall** [comment]
Yes - I may not be rememberign quite right, but I did the research when I was writing that code within chapter 5, so that is going to be correct.

**Comment by ericbrunner** [comment]
Ok thanks. I looked into the REST API docs and can't find that URI `$"/push/installations/{client.InstallationId}`. Can you explain me where you have that from? 

At: https://msdn.microsoft.com/en-us/library/azure/mt621153.aspx

the Request has that format

`https://{namespace}.servicebus.windows.net/{NotificationHub}/installations/{id}?api-version=2015-01`

**Comment by adrianhall** [comment]
The /push/installations/{client.InstallationId} is the endpoint for App Service Push.  It isn't part of NH.  It's a specific piece of code that translates what the ZUMO SDK provides into the NH equivalent, after dealing with the tag settings.  So the full URI is `https://{yoursite}.azurewebsites.net/push/installations/{client.InstallationId}`

**Comment by adrianhall** [comment]
Added some additional callouts in chapter 5 to fix the confusion.  Fixed in 28ca554

**Comment by ericbrunner** [comment]
@adrianhall Great. Thanks for clarification
